MCS-1187 Separate lava property in scene files (Python)

ThomasSchellenbergNextCentury opened the pull request:
Previously, we used the `floorTextures` scene property with a specific "lava" material to represent lava. While this made sense early on, when lava was simply a texture we applied to some areas of the floor, now that lava's become more complex (counting steps in lava, setting reward penalties, automatically ending the scene, etc.), it makes sense as its own separate scene file property. Therefore I've added `lava` to the SceneConfiguration/SceneConfigurationSchema classes.

I'll merge Jacob's lava grid changes into here if we decide to move forward with them.

Please note: branching off https://github.com/NextCenturyCorporation/MCS/pull/514

Corresponding Unity PR: https://github.com/NextCenturyCorporation/ai2thor/pull/209
Corresponding ILE PR: Coming soon

TODOs:
- I want to add some lava integration tests, but will wait until we make a final decision on Jacob's lava grid change PR.

ThomasSchellenbergNextCentury commented on machine_common_sense/material.py:
```diff
@@ -72,6 +72,3 @@ def verify_material_enum_string(enum_string):
             return False
-
-
-LAVA_TEXTURES = ["Stylized Lava Texture/Materials/Stylize_Lava_diffuse"]
```
We're no longer dependent on maintaining all of the lava textures in this codebase.

kdrumm replied:
Yay!  If I understand this all correctly, only the unity repo will need the list of textures, right?

ThomasSchellenbergNextCentury replied:
That's correct! I should be able to remove them from the scene generator code as well.

ThomasSchellenbergNextCentury commented on machine_common_sense/step_metadata.py:
```diff
@@ -73,15 +73,17 @@ class StepMetadata:
-        if using a metadata level below the 'oracle' level.
+    holes : list of tuples
+        Coordinates of holes as (X, Z) float tuples. Will be set to 'None' if
+        using a metadata level below the 'oracle' level.
     image_list : list of Pillow.Image objects
         The list of images from the scene after the last action and physics
         simulation were run. This is usually a list with 1 image, except for
         the output from start_scene for a scene with a scripted Preview Phase.
         (Preview Phase case details TBD).
-    lava: list of Vector2dInt - coordinates of lava.  Will be set to 'None'
-        if using a metadata level below the 'oracle' level.
+    lava : list of tuples
+        Coordinates of pools of lava as (X, Z) float tuples. Will be set to
+        'None' if using a metadata level below the 'oracle' level.
```
The `holes` and `lava` were being returned as Vector2dInt, but that seemed weird, since we're not returning config manager classes in any of the other metadata properties, so I changed these to lists of tuples. If we want to start returning config manager classes in metadata properties (not just holes/lava), I'd recommend we look at it in a separate ticket.

ThomasSchellenbergNextCentury commented on tests/test_controller_output_handler.py:
```diff
@@ -458,7 +459,11 @@ def test_wrap_output(self):
         mock_event = self.create_mock_scene_event(mock_scene_event_data)
 
         coh = ControllerOutputHandler(self._config)
-        coh.set_scene_config(SceneConfiguration((mock_scene_event_data)))
+        coh.set_scene_config(SceneConfiguration(holes=[
+            Vector2dInt(x=0, z=0), Vector2dInt(x=1, z=2), Vector2dInt(x=9, z=8)
+        ], lava=[
+            Vector2dInt(x=3, z=3), Vector2dInt(x=7, z=5), Vector2dInt(x=4, z=6)
+        ]))
```
We weren't creating the SceneConfiguration correctly in a couple of places in this test file -- the `mock_scene_event_data` is the (mock) payload sent from Unity, which doesn't impact the SceneConfiguration at all. (It wasn't breaking any tests; it just wasn't doing anything.) So I removed `mock_scene_event_data` from the SceneConfiguration constructor, and then added holes and lava to test those conditions for this PR.

ThomasSchellenbergNextCentury commented on tests/test_step_metadata.py:
```diff
@@ -127,15 +127,15 @@ def test_head_tilt(self):
     def test_holes(self):
-        self.assertEquals(self.step_metadata.holes, [])
+        self.assertEqual(self.step_metadata.holes, [])
         self.assertIsInstance(self.step_metadata.holes, list)
 
     def test_image_list(self):
         self.assertFalse(self.step_metadata.image_list)
         self.assertIsInstance(self.step_metadata.image_list, list)
 
     def test_lava(self):
-        self.assertEquals(self.step_metadata.lava, [])
+        self.assertEqual(self.step_metadata.lava, [])
```
Fixed a deprecation warning.

deanwetherby reviewed (approved):
I like that this is being explicit about lava.

jaudick commented:
@ThomasSchellenbergNextCentury I think the test_plotter lava test should be updated to match this new config

ThomasSchellenbergNextCentury commented:
I added two new integration tests for lava (107 and 108), since we didn't previously have any, and also updated the existing integration tests for holes and floor textures, so they all verify the `holes` and `lava` oracle output in the step metadata.

ThomasSchellenbergNextCentury commented:
@jaudick Thanks for alerting me about the plotter code. It's now been updated. We'll have to coordinate when merging our branches since my changes will likely cause conflicts with your changes.

https://github.com/NextCenturyCorporation/MCS/pull/518/commits/2b8b036975bcad72977892a66bad5842f0c7b775

jaudick commented:
@ThomasSchellenbergNextCentury I'm ok just merging yours into development first and holding off on merging mine until then. I'll resolve the conflicts on my end, it shouldn't be too difficult

jaudick reviewed (approved):
Integration tests pass for me
